lint: Address a few idiomatic Go improvements

simar7 opened the pull request:
Signed-off-by: Simarpreet Singh <simar@linux.com>

itaysk commented on tracee/tracee.go:
```diff
@@ -765,7 +765,7 @@ func (t *Tracee) initBPF(bpfObjectPath string) error {
 
 	supportRawTracepoints, err := supportRawTP()
 	if err != nil {
-		return fmt.Errorf("Failed to find kernel version: %v", err)
```
The errors that I write are usually lowercased, as mentioned here: https://github.com/golang/go/wiki/CodeReviewComments#error-strings

we do have places where errors are mistakenly styles like a sentence, if you want you could change those instead

simar7 replied:
Hmm I didn't understand your comment, are you saying this is more of a log statement than an error itself? So in terms of code is your suggestion:

```
log.Printf("Failed to find kernel version: %v", err)
return err
```

itaysk replied:
again, read this the wrong way :)

itaysk commented on tracee/tracee_test.go:
```diff
@@ -120,16 +120,40 @@ func TestReadArgFromBuff(t *testing.T) {
+		_, actual, err := readArgFromBuff(b)
 		assert.Equal(t, tc.expectedError, err, tc.name)
 		assert.Equal(t, tc.expectedArg, actual, tc.name)
+
+		// TODO: the following fails for:
+		// strArrT
+		// unknown
+		// sockAddrT - AF_UNIX
```
can you please open an issue for this instead of code comment?

simar7 replied:
Yes https://github.com/aquasecurity/tracee/issues/451

itaysk replied:
thanks, so just remove the comment?

simar7 replied:
Right - https://github.com/aquasecurity/tracee/pull/427/commits/3e460e1240b2cb6daae2a6779c4375f16c352874

itaysk commented on tracee/tracee.go:
```diff
@@ -638,19 +638,19 @@ func (t *Tracee) populateBPFMaps() error {
 	if err != nil {
 		return fmt.Errorf("error getting BPF program trace_ret_vfs_write_tail: %v", err)
 	}
-	bpfProgArrayMap.Update(uint32(tailVfsWrite), uint32(prog.GetFd()))
```
can you please explain this suggestion? `tailVfsWrite` is a uint32 anyway

simar7 replied:
Yes exactly - it's an uint32 already so this cast is unnecessary.

itaysk replied:
oh right, I read it the other way

itaysk replied:
read this the wrong way :) thanks

yanivagman commented:
What about the failed test?

yanivagman reviewed (changes requested):
We shouldn't merge a commit which cause the tests to fail

itaysk commented on tracee/tracee_test.go:
```diff
@@ -120,16 +120,40 @@ func TestReadArgFromBuff(t *testing.T) {
+			input: []byte{12, //sockAddrT
+				0,    // Dummy tag
+				1, 0, //sa_family=AF_UNIX
+				47, 116, 109, 112, 47, 115, 111, 99, 107, 101, 116, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 101, 110, 0, 0, 0, // sun_path=/tmp/socket
```
the max length of `sun_path` is 108 bytes. here you give 114 bytes, which explains the extra 6 bytes that fails the test.

simar7 replied:
Ah nice catch - you're right.

itaysk commented on tracee/tracee_test.go:
```diff
@@ -120,16 +120,40 @@ func TestReadArgFromBuff(t *testing.T) {
+		// strArrT
+		// unknown
+		// sockAddrT - AF_UNIX
+		assert.Empty(t, b.Len(), tc.name) // passed in buffer should be emptied out
```
In the case of unknown, we can't consume the event from the buffer since we don't know the event's size. (we should stop tracee in this case). so I don't think we should assert for emptiness in this case

simar7 replied:
Does that happen today? (The stoppage of Tracee that is) I just see all errors are collected and printed via a handler.

simar7 replied:
Anyways - that's tangential to this PR, we can take it offline.

yanivagman commented:
I think it is better to split this into 2 PRs - one for lint and go improvements, and the other for adding/changing tests

grantseltzer commented:
@simar7 It seems like this just needs a rebase. 

@yanivagman In general I agree with you but for this case I think it's small enough to let it slide.

simar7 commented:
> I think it is better to split this into 2 PRs - one for lint and go improvements, and the other for adding/changing tests

I agree @yanivagman - I shouldn't have coupled those two. Will keep that in mind for the next one!

simar7 commented:
> @simar7 It seems like this just needs a rebase.
> 
> @yanivagman In general I agree with you but for this case I think it's small enough to let it slide.

Just updated and rebased - PTAL

itaysk commented:
Thanks @simar7 , LGTM, but if we are fixing error capitalization, can you please fix the rest as well? a quick scan revealed more:
```
❯ grep -rn -e 'fmt.Errorf("[A-Z]' .
./libbpfgo/libbpfgo.go:312:             return fmt.Errorf("Failed to pin map %s to path %s", b.name, pinPath)
./libbpfgo/libbpfgo.go:324:             return fmt.Errorf("Failed to unpin map %s from path %s", b.name, pinPath)
./libbpfgo/libbpfgo.go:336:             return fmt.Errorf("Failed to set pin for map %s to path %s", b.name, pinPath)
./libbpfgo/libbpfgo.go:358:             return nil, fmt.Errorf("Unknown data type %T", data)
./libbpfgo/libbpfgo.go:648:                             return fmt.Errorf("Error polling perf buffer: %d", err)
./tracee-rules/signatures/rego/regosig/traceerego.go:150:       return fmt.Errorf("OnSignal is not implemented")
./tracee-ebpf/main.go:73:                               return fmt.Errorf("Insufficient privileges to run")
./tracee-ebpf/main.go:1178:             return fmt.Errorf("Failed to make BPF object (clang): %v. Try using --debug for more info", err)
./tracee-ebpf/main.go:1197:             return fmt.Errorf("Failed to make BPF object (llc): %v. Try using --debug for more info", err)
./tracee-ebpf/main.go:1214:                     return fmt.Errorf("Failed to make BPF object (llvm-strip): %v. Try using --debug for more info", err)
./tracee-ebpf/tracee/pipeline.go:165:                                   errc <- fmt.Errorf("Invalid arg tag for event %d", rawEvent.Ctx.EventID)
./tracee-ebpf/tracee/tracee.go:162:                     return fmt.Errorf("The length of a path filter is limited to 50 characters: %s", filter)
./tracee-ebpf/tracee/tracee.go:768:             return fmt.Errorf("Failed to find kernel version: %v", err)
```

(up to you if fix the ones in other projects in same commit or not)

itaysk commented:
I also looked for errors ending with period and found just one in libbpfgo:

```
❯ grep -rn -e 'fmt.Errorf(".*\."' .  
./libbpfgo/libbpfgo.go:253:             return nil, fmt.Errorf("failed to open BPF object %s: %v...", bpfObjName, bpfObjBuff[:20])
```

simar7 commented:
> Thanks @simar7 , LGTM, but if we are fixing error capitalization can you please fix the rest as well? a quick scan revealed more:
> 
> ```
> ❯ grep -rn -e 'fmt.Errorf("[A-Z]' .
> ./libbpfgo/libbpfgo.go:312:             return fmt.Errorf("Failed to pin map %s to path %s", b.name, pinPath)
> ./libbpfgo/libbpfgo.go:324:             return fmt.Errorf("Failed to unpin map %s from path %s", b.name, pinPath)
> ./libbpfgo/libbpfgo.go:336:             return fmt.Errorf("Failed to set pin for map %s to path %s", b.name, pinPath)
> ./libbpfgo/libbpfgo.go:358:             return nil, fmt.Errorf("Unknown data type %T", data)
> ./libbpfgo/libbpfgo.go:648:                             return fmt.Errorf("Error polling perf buffer: %d", err)
> ./tracee-rules/signatures/rego/regosig/traceerego.go:150:       return fmt.Errorf("OnSignal is not implemented")
> ./tracee-ebpf/main.go:73:                               return fmt.Errorf("Insufficient privileges to run")
> ./tracee-ebpf/main.go:1178:             return fmt.Errorf("Failed to make BPF object (clang): %v. Try using --debug for more info", err)
> ./tracee-ebpf/main.go:1197:             return fmt.Errorf("Failed to make BPF object (llc): %v. Try using --debug for more info", err)
> ./tracee-ebpf/main.go:1214:                     return fmt.Errorf("Failed to make BPF object (llvm-strip): %v. Try using --debug for more info", err)
> ./tracee-ebpf/tracee/pipeline.go:165:                                   errc <- fmt.Errorf("Invalid arg tag for event %d", rawEvent.Ctx.EventID)
> ./tracee-ebpf/tracee/tracee.go:162:                     return fmt.Errorf("The length of a path filter is limited to 50 characters: %s", filter)
> ./tracee-ebpf/tracee/tracee.go:768:             return fmt.Errorf("Failed to find kernel version: %v", err)
> ```
> 
> (up to you if fix the ones in other projects in same commit or not)

OK - I have addressed them here https://github.com/aquasecurity/tracee/pull/427/commits/07bc04fc168561fa6ef3510205c3f497ad1d55d1

yanivagman commented on tracee-ebpf/tracee/tracee.go:
```diff
@@ -1345,13 +1345,13 @@ func (t *Tracee) processFileWrites() {
 				}
 			} else {
-				if _, err := f.Seek(int64(meta.Off), os.SEEK_SET); err != nil {
+				if _, err := f.Seek(int64(meta.Off), io.SeekStart); err != nil {
```
This needs to be seek_set so tracee can rebuild the file according to the correct offsets. If you set it to seekstart instead, it will overwrite other data written, and the written file will be corrupted

itaysk replied:
@yanivagman `os.SEEK_SET` is equivalent to `io.SeekStart`. This change is merely using the same value from a different package (which is good)

yanivagman replied:
Correct. My mistake.

yanivagman reviewed (changes requested):
Please see my comment about seek_set
